fs/operations/operations.go

```diff
@@ -1026,6 +1028,17 @@ func HashLister(ctx context.Context, ht hash.Type, f fs.Fs, w io.Writer) error {
 	})
 }
 
+// HashListerBase64 does a md5sum equivalent for the hash type passed in with base64 encoded
+func HashListerBase64(ctx context.Context, ht hash.Type, f fs.Fs, w io.Writer) error {
+	return ListFn(ctx, f, func(o fs.Object) {
+		sum := hashSum(ctx, ht, o)
```

Comment:
> Note that `sum` can be "UNSUPPORTED" or "ERROR" here.
> 
> I think I would alter `hashSum` to return (string, error) then
> 
> ```go
> sum, err := hashSum(ctx, ht, o)
> if err == nil {
>     hexBytes, _ := hex.DecodeString(sum)
>     sum = base64.URLEncoding.EncodeToString(hexBytes)
> }
> ```

should I add this feature? Is this a breaking change in Golang?

Reply:
In Golang, the error handling is often done by returning an Error object instead of raising an exception. So if `err` is not nil, you know there is something wrong here.

It's not a feature but more like a historical problem in Golang.

Please refer to https://medium.com/rungo/error-handling-in-go-f0125de052f0 and https://blog.golang.org/error-handling-and-go to know more about error handling in golang.

I think what @ncw said is to modify the original `hashSum` function and return errors instead of giving "UNSUPPORTED" or "ERROR", so the caller can catch the error and not continue working. Correct me if I am wrong.

Reply:
what I want to ask is whether it will cause some trouble if I add the second return value or it is just transparent to the other code. 
however I find this function is only used by the method I am modifying so I think it will not cause any trouble.
Some behaviors are odd in Golang. In other language, it will return a false to the caller and set a referenced input param to false.